engine/classes/Elgg/Application/ServeFileHandler.php

```diff
@@ -106,11 +108,10 @@ public function getResponse(Request $request) {
 		$response = new BinaryFileResponse($filenameonfilestore, 200, $headers, $public, $content_disposition);
 		$response->prepare($request);
 
-		if (empty($expires)) {
-			$expires = strtotime('+1 year');
+		if ($expires) {
+			$expires_dt = (new DateTime())->setTimestamp($expires);
+			$response->setExpires($expires_dt);
 		}
```

Comment:
As per spec expires shouldn't be over 1 year, but it seems that browsers don't care. In ckeditor_addons, I used +99999 years :)